src/ensembl/src/shared/types/utility-types/modify.ts

```diff
@@ -0,0 +1,22 @@
+/*
+
+The Modify type makes it possible to overwrite specific fields of a given type
+
+e.g.:
+
+type InitialType = {
+  foo: string;
+  bar: string;
+};
+
+type ModifiedType = Modify<InitialType, {
+  bar: number;
+}>
+
+will give us the ModifiedType that is {
+  foo: string,
+  bar: number
+}
+*/
+
+export type Modify<T, R> = Omit<T, keyof R> & R;
```

Comment:
Is this used anywhere?

Reply:
Not at the moment, no; I thought I needed it, but then updated the code so that I didn't. But it is a useful utility, which would be a shame to google for next time we might need it. I thought I might just as well keep it in our repo. I can remove it if you want.

Reply:
Let's keep it 👍